app/models/miq_filter.rb

```diff
@@ -174,8 +174,12 @@ def self.apply_belongsto_filters(inputs, bfilters)
           end
           break if filtered.last == p
         else
-          vcmeta = belongsto2object(tag)
+          vcmeta_list = belongsto2object_list(tag)
+          vcmeta_list.pop if vcmeta_list.last.kind_of?(Host)
+          vcmeta = vcmeta_list.last
```

Comment:
@jrafanie I added spec for it. (https://github.com/ManageIQ/manageiq/commit/a71631edaae75c2196b5d69ee056899369170581)